Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift

```diff
@@ -123,88 +104,50 @@ struct ChannelDetailPayload: Decodable {
-        truncatedAt = try container.decodeIfPresent(Date.self, forKey: .truncatedAt)
-        createdBy = try container.decodeIfPresent(UserPayload.self, forKey: .createdBy)
-        // For `hidden`, we don't fallback to `false`
-        // since this field is not sent for all API calls and for events
-        // We can't assume anything regarding this flag when it's absent
-        isHidden = try container.decodeIfPresent(Bool.self, forKey: .hidden)
-        lastMessageAt = try container.decodeIfPresent(Date.self, forKey: .lastMessageAt)
-        isFrozen = try container.decode(Bool.self, forKey: .frozen)
-        team = try container.decodeIfPresent(String.self, forKey: .team)
-        memberCount = try container.decodeIfPresent(Int.self, forKey: .memberCount) ?? 0
-        
-        members = try container.decodeIfPresent([MemberPayload].self, forKey: .members)
-        
-        cooldownDuration = try container.decodeIfPresent(Int.self, forKey: .cooldownDuration) ?? 0
-        
+         
+        let extraData: [String: RawJSON]
         if var payload = try? [String: RawJSON](from: decoder) {
             payload.removeValues(forKeys: ChannelCodingKeys.allCases.map(\.rawValue))
             extraData = payload
         } else {
             extraData = [:]
         }
-    }
-    
-    // MARK: - For testing
-    
-    internal init(
-        cid: ChannelId,
-        name: String?,
-        imageURL: URL?,
-        extraData: [String: RawJSON],
-        typeRawValue: String,
-        lastMessageAt: Date?,
-        createdAt: Date,
-        deletedAt: Date?,
-        updatedAt: Date,
-        truncatedAt: Date?,
-        isHidden: Bool? = nil,
-        createdBy: UserPayload?,
-        config: ChannelConfig,
-        isFrozen: Bool,
-        memberCount: Int,
-        team: String?,
-        members: [MemberPayload]?,
-        cooldownDuration: Int
-    ) {
-        self.cid = cid
-        self.name = name
-        self.imageURL = imageURL
-        self.extraData = extraData
-        self.typeRawValue = typeRawValue
-        self.lastMessageAt = lastMessageAt
-        self.createdAt = createdAt
-        self.deletedAt = deletedAt
-        self.updatedAt = updatedAt
-        self.truncatedAt = truncatedAt
-        self.createdBy = createdBy
-        self.config = config
-        self.isFrozen = isFrozen
-        self.memberCount = memberCount
-        self.team = team
-        self.members = members
-        self.cooldownDuration = cooldownDuration
-        self.isHidden = isHidden
+        
+        self.init(
+            cid: try container.decode(ChannelId.self, forKey: .cid),
+            name: try container.decodeIfPresent(String.self, forKey: .name),
+            // Unfortunately, the built-in URL decoder fails, if the string is empty. We need to
+            // provide custom decoding to handle URL? as expected.
+            imageURL: try container.decodeIfPresent(String.self, forKey: .imageURL).flatMap(URL.init(string:)),
```

Comment:
@evsaev Maybe we could leverage the power of property wrappers here.

Look [here](https://youtu.be/cw95S8gwkEU?t=1295).

Reply:
Yep, that's a good idea but if you don't mind I leave it as it is for now. The refactoring here was to not write memberwise `init` manually only for tests but to use the generated one. I didn't change anything else here to not increase the scope.

Reply:
ok fair enough. can you please create tech item for this improvement?

Reply:
Yep, now there's one in JIRA 👍Thanks for pointing this out, I feel that sometimes a custom `init(decoder:)` is needed just because of the URL decoding issue